node/src/neighborhood/mod.rs

```diff
@@ -3325,21 +3399,36 @@ mod tests {
         join_rows(db, (&k, &l, &m, &n, &o), (&p, &q, &r, &s, &t));
         join_rows(db, (&p, &q, &r, &s, &t), (&u, &v, &w, &x, &y));
         designate_root_node(db, &l);
-        let before = Instant::now();
-
-        // All the target-designated routes from L to N
-        let route = subject
-            .find_best_route_segment(&l, Some(&n), 3, 10000, RouteDirection::Back, None)
-            .unwrap();
+        let limit_ms = 50;
+        // Sometimes a system delay might run this over the limit. Give it three chances.
+        let total_chances = 3;
+        let mut retries_left = total_chances - 1;
+        let mut intervals: Vec<u128> = vec![];
+        let route = loop {
+            let before = Instant::now();
+
+            // All the target-designated routes from L to N
+            let route = subject
+                .find_best_route_segment(&l, Some(&n), 3, 10000, RouteDirection::Back, None)
+                .unwrap();
 
-        let after = Instant::now();
+            let after = Instant::now();
+            let interval = after.duration_since(before);
+            if interval.as_millis() > limit_ms {
+                intervals.push(interval.as_millis());
+                if retries_left == 0 {
+                    panic!(
+                        "Should have calculated route in <={}ms, but failed after {} tries: {:?}",
+                        limit_ms, total_chances, intervals
+                    )
+                } else {
+                    retries_left -= 1
+                }
+            } else {
+                break route
+            }
+        };
         assert_eq!(route, vec![&l, &g, &h, &i, &n]); // Cheaper than [&l, &q, &r, &s, &n]
-        let interval = after.duration_since(before);
-        assert!(
-            interval.as_millis() <= 100,
```

Comment:
I don't remember precisely, but as I think about it now, it seems to me that a tenth of a second is a long, long time for calculating a route, since we need to calculate routes so often. I think it's reasonable that if we can't get the route-calculation time under 50ms in three tries, we need to know about it so that we can look at optimizing the algorithm.